### DISCUSSION TITLE
Is it possible to use WebDriver to simply compare two images for visual regression?

### DISCUSSION THREAD

**Discussion by good-idea** [webdriverio/webdriverio#13559] [Q&A]
I'm working on a React Native project that is used on a custom ROM, so we aren't able to use a lot of the existing visual regression frameworks & platforms out there. We'll write our own scripts to get the baseline & comparison images - so we're just looking for something that will allow us to compare them. It looks like all of the visual snapshot matching methods leverage `browser` to get the snapshots.

Would it be possible to write a custom matcher that is simply `expect('/baseline/image-one.png').toMatch('/comparison/image-one.png')`?

**Comment by christian-bromann** [comment]
@good-idea WebdriverIO has built-in support for visual regression testing and image comparison, find more information [in the project docs](https://webdriver.io/docs/visual-testing).

**Reply by good-idea** [reply]
@christian-bromann thank you for your reply. I've looked through the docs, and like I mentioned above, capturing and comparing screenshots happens through using `browser.toMatchScreenshot(...)` and other similar methods, which won't work for our project. I was hoping to find a lower-level method to just pass in two filenames, and, if there is a diff, export the diffed image.

**Reply by christian-bromann** [reply]
WebdriverIO provides [such methods](https://webdriver.io/docs/visual-testing/methods/#check-methods) as well.
